### PR TITLE
SimpleXmlBuilder

### PR DESCRIPTION
**Pull request by CyberSecutor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,176 @@
...
+* Modified asXml() to accept an extra parameter for outputting the long 
+format, increasing readability.
+
+```php
+    public string|boolean SimpleXMLElement::asXML (
+        [ string $filename ] [, boolean $longOutput ]
+    )
+```
+
+* Using a associative PHP array to generate an XML message.
```
Comment (a-schuurman) [inline README.md]: Using **an** associative..

File: src/SimpleXmlBuilder.php [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                            htmlentities($values, ENT_QUOTES),
+                            $namespace
+                        );
+                    }
+
+                }
+            }
+        }
+
+        // Returns null if $document was empty.
```
Comment (a-schuurman) [inline src/SimpleXmlBuilder.php:105]: Should be `$xmlDocument`.

File: src/SimpleXmlBuilder.php [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                    };
+                } elseif (is_scalar($values)) {
+                    if (strpbrk($values, static::$dangerousCharacters)) {
+                        /** @var SimpleXmlBuilder $child */
+                        $child = $xmlDocument->addChild($element, null, $namespace);
+                        $child->addCData($values);
+                    } else {
+                        $xmlDocument->addChild(
+                            $element,
+                            htmlentities($values, ENT_QUOTES),
```
Comment (a-schuurman) [inline src/SimpleXmlBuilder.php]: Perhaps `htmlspecialchars`?

File: src/SimpleXmlBuilder.php [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        $node->appendChild($no->createCDATASection($nodeValue));
+
+        return true;
+    }
+
+    /**
+     * Return a well-formed XML string based on SimpleXML element.
+     *
+     * @param string $filename Optional filename.
+     * @param boolean $longOutput .
```
Comment (a-schuurman) [inline src/SimpleXmlBuilder.php]: Lonely dot at the end of the line.

File: src/SimpleXmlBuilder.php (hunk 2 above)
Comment (CyberSecutor) [inline src/SimpleXmlBuilder.php:105]: No, it returns null if the `$document` parameter was an empty array.

Comment (a-schuurman) [inline src/SimpleXmlBuilder.php]: :mask:

**Pull request merged**